nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,524 @@
+
+    dataDir = mkOption {
+      type = path;
+      default = "/var/lib/sympa";
+      description = "Data directory where state is stored.";
+    };
+
+    user = mkOption {
+      type = str;
+      default = "sympa";
+      description = "What to call the Sympa user (must be used only for sympa).";
+    };
+
+    group = mkOption {
+      type = str;
+      default = "sympa";
+      description = "What to call the Sympa group (must be used only for sympa).";
+    };
+
+    lang = mkOption {
+      type = str;
+      default = "en_US";
+      example = "cs";
+      description = "Sympa language.";
+    };
+
+    listMasters = mkOption {
+      type = listOf str;
+      example = [ "postmaster@sympa.example.org" ];
+      description = ''
+        The list of the email addresses of the listmasters
+        (users authorized to perform global server commands).
+      '';
+    };
+
+    mainDomain = mkOption {
+      type = nullOr str;
+      default = null;
+      example = "lists.example.org";
+      description = ''
+        Main domain to be used in sympa.conf.
+        If null, one of the <option>domains</option> is chosen for you.
+      '';
+    };
+
+    domains = mkOption {
+      type = attrsOf (submodule ({ name, config, ... }: {
+        options = {
+          webHost = mkOption {
+            type = nullOr str;
+            default = null;
+            example = "archive.example.org";
+            description = ''
+              Domain part of the web interface URL (no web interface for this domain if null).
+              DNS record of type A (or AAAA) has to exist with this value.
+            '';
+          };
+          webLocation = mkOption {
+            type = str;
+            default = "/";
+            example = "/sympa";
+            description = "URL path part of the web interface.";
+          };
+          settings = mkOption {
+            type = attrsOf (either (either str int) bool);
+            default = {};
+            example = {
+              default_max_list_members = 3;
+            };
+            description = ''
+              The robot.conf configuration file as key value set.
+            '';
+          };
+        };
+
+        config.settings = mkIf (cfg.web.enable && config.webHost != null) {
+          wwsympa_url = mkDefault "https://${config.webHost}${strings.removeSuffix "/" config.webLocation}";
+        };
+      }));
+
+      description = ''
+        Email domains handled by this instance. There have
+        to be MX records for keys of this attribute set.
+      '';
+      example = literalExample ''
+        {
+          "lists.example.org" = {
+            webHost = "lists.example.org";
+            webLocation = "/";
+          };
+          "sympa.example.com" = {
+            webHost = "example.com";
+            webLocation = "/sympa";
+          };
+        };
+      '';
+    };
+
+    database = {
+      type = mkOption {
```

Comment:
Ping @aanderse for this option set, I'm not sure how this interacts with `ensureDatabases`

Reply:
Seems to be working now, PTAL.